contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,781 @@
+# Scheduling Policy
+
+_Status: Draft_
+
+_Authors: @arnaudmz, @yastij_
+
+_Reviewers: @bsalamat, @liggitt_
+
+# Objectives
+
+-  Define the concept of scheduling policies
+-  Propose their initial design and scope
+
+## Non-Goals
+
+-  How taints / tolerations work
+-  How NodeSelector works
+-  How node / pod affinity / anti-affinity rules work
+-  How several schedulers can be used within a single cluster
+-  How priority classes work
+
+# Background
+
+During real-life Kubernetes architecturing we encountered contexts where role-isolation (between administration and simple namespace usage in a multi-tenant context) could be improved. So far, no restriction is possible on toleration, priority class usage, nodeSelector, anti-affinity depending on user permissions (RBAC).
+
+Identified use-cases aim to ensure that administrators have a way to restrict users or namepace when
```

Comment:
These aren't great use cases - I really expected a more end kubernetes user focus:

1. Allow administrators to restrict execution for specific applications (which are namespace scoped) into certain nodes
2. Allow administrators to create policies that prevent users from even attempting to schedule workloads onto masters to maximize security

etc.

Something as critical as policy needs a lot more use case design before we even get into implementation details.

Reply:
I'm probably not going to be happy until at least 300 lines of this doc is a detailed justification for the design space and what we are actually trying to build.  If that exists elsewhere, please link it here.

Reply:
@smarterclayton - Indeed there's some another use case I've got with @arnaudmz, we'll add them. I'll try to put up a design section to have an overview about this policy, SGTY ?

Reply:
See also discussions in:

#17190, #17151, #17097, #14573

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/taint-toleration-dedicated.md